File: src/main/java/cn/edu/tsinghua/sdfs/protocol/serilizer/impl/JSONSerializer.java
package cn.edu.tsinghua.sdfs.protocol.serilizer.impl;


import cn.edu.tsinghua.sdfs.protocol.serilizer.Serializer;
import com.alibaba.fastjson.JSON;

public class JSONSerializer implements Serializer {

    @Override
    public byte[] serialize(Object object) {
        return JSON.toJSONBytes(object);
    }

    @Override
    public <T> T deserialize(byte[] bytes, Class<? extends T> clazz) {
        return JSON.parseObject(bytes, clazz);
    }
}

File: src/main/java/cn/edu/tsinghua/sdfs/protocol/serilizer/impl/JSONSerializer.kt
package cn.edu.tsinghua.sdfs.protocol.serilizer.impl


import cn.edu.tsinghua.sdfs.protocol.serilizer.Serializer
import com.alibaba.fastjson.JSON

class JSONSerializer : Serializer {

    override fun serialize(obj: Any): ByteArray {
        return JSON.toJSONBytes(obj)
    }

    override fun <T> deserialize(bytes: ByteArray, clazz: Class<out T>): T {
        return JSON.parseObject(bytes, clazz)
    }
}
